### PR TITLE
Upgrades and experimental api support

### PR DESCRIPTION
**Pull request by losolio**
- [x] Update 404 and 500 pages
- [x] Upgrade wagtail and backend packages
- [x] Add experimental support for rss, rest api and graphql
- [x] Update front end packages
- [x] Accesibility updates

### CODE REVIEW

File: website/articles/models.py [hunk 1]
```diff
@@ -118,6 +121,26 @@ class ArticlePage(BasePage):
...
     ]
 
+    # Export fields over REST API
+    api_fields = [
+        APIField('introduction'),
+        APIField('body'),
+        APIField('authors'),
+        APIField('tags'),
+        APIField('categories'),
+        APIField('featured_image', serializer=ImageRenditionField('fill-1920x1080')),
```
Comment (tombola) [inline website/articles/models.py:131]: The ImageRenditionField may need an explicit `source` keyword argument:

https://docs.wagtail.io/en/v2.6.3/advanced_topics/api/v2/configuration.html#images-in-the-api

File: website/articles/models.py [hunk 2]
```diff
@@ -118,6 +121,26 @@ class ArticlePage(BasePage):
...
+    ]
+
+    # Export fields to GraphQL
+
+    graphql_fields = [
+        GraphQLString("heading"),
+        GraphQLString("publication_date"),
+        GraphQLStreamfield("body"),
+        GraphQLString("categories"),
+        GraphQLString("featured_image")
```
Comment (tombola) [inline website/articles/models.py:141]: This should probably be `GraphQLImage`

File: website/news/models.py [hunk 3]
```diff
@@ -100,6 +103,12 @@ class NewsPage(BasePage):
         FieldPanel('publication_date'),
     ]
 
+    graphql_fields = [
+        GraphQLString("summary"),
+        GraphQLString("body"),
+        GraphQLString("featured_image")
```
Comment (tombola) [inline website/news/models.py:109]: This should use GraphQLImage.

File: website/people/models.py [hunk 4]
```diff
@@ -75,6 +76,14 @@ class PersonPage(BasePage):
         StreamFieldPanel('biography')
     ]
 
+    graphql_fields = [
+        GraphQLString("name"),
+        GraphQLString("job_title"),
+        GraphQLString("introduction"),
+        GraphQLString("photo"),
```
Comment (tombola) [inline website/people/models.py:83]: This should use GraphQLImage

File: website/articles/models.py (hunk 1 above)
Comment (losolio) [reply website/articles/models.py:131]: Thanks, @tombola - The API part of this is highly experimental, and this does not seem to break the solution (if it breaks - just remove the line). Will look into this for next release.

File: website/project_styleguide/templates/patterns/base.html [hunk 5]
```diff
@@ -35,6 +35,14 @@
         <script src="{% static 'js/main.js' %}"></script>
 
         {% block extra_js %}{% endblock %}
+        <script>
+            function selectSearchBox() {
+                setTimeout(() => {
+                    document.getElementById("navbarSearch").focus();
```
Comment (tombola) [inline website/project_styleguide/templates/patterns/base.html:41]: I am not certain of the accessibility implications of enforcing focus on an element directly after page load.

**Review by tombola — COMMENTED** [review]
I have left a couple of minor comments regarding graphql field usage, and an accessibility concern.

File: website/utils/models.py [hunk 6]
```diff
@@ -193,6 +193,9 @@ class BasePage(SocialFields, ListingFields, Page):
     class Meta:
         abstract = True
 
+    def get_absolute_url(self):
```
Comment (tombola) [inline website/utils/models.py:196]: I cannot see this in use elsewhere, does it have an intended use case?

**Comment by losolio** [comment]
Thanks for your review, @tombola - I have left some comments in the ticket on codebase.

**Pull request merged**